sigopt/xgboost/run.py

```diff
@@ -0,0 +1,109 @@
+import platform
+import copy
+
+import xgboost
+# pylint: disable=no-name-in-module
+from xgboost import DMatrix
+
+from ..context import Context
+from ..log_capture import SystemOutputStreamMonitor
+from .. import create_run
+
+DEFAULT_EVALS_NAME = 'Test Set'
+DEFAULT_RUN_OPTIONS = {
+  'log_sys_info': True,
+  'log_stdout': True,
+  'log_stderr': True,
+  'run': None
+}
+XGB_ALIASES = \
+  {
+     'learning_rate': 'eta',
+     'min_split_loss': 'gamma',
+     'reg_lambda': 'lambda',
+     'reg_alpha': 'alpha'
+  }
+
+
+def parse_run_options(run_options):
+  if run_options:
+    assert run_options.keys() <= DEFAULT_RUN_OPTIONS.keys(), 'Unsupported argument inside run_options'
+  run_options_parsed = {**DEFAULT_RUN_OPTIONS, **run_options} if run_options else DEFAULT_RUN_OPTIONS
+  return run_options_parsed
+
+
+class XGBRun:
+
+  def __init__(self, params, dtrain, num_boost_round, evals, run_options):
+    self.params = params
+    self.dtrain = dtrain
+    self.num_boost_round = num_boost_round
+    self.evals = evals
+    self.run_options_parsed = parse_run_options(run_options)
+    self.validation_sets = [(evals, DEFAULT_EVALS_NAME)] if isinstance(evals, DMatrix) else evals
+    self.run = None
+
+  def make_run(self):
+    if self.run_options_parsed['run']:
+      self.run = self.run_options_parsed['run']
+    else:
+      self.run = create_run()
+
+  def log_metadata(self):
+    if self.run_options_parsed['log_sys_info']:
+      python_version = platform.python_version()
+      self.run.log_metadata("Python Version", python_version)
+      self.run.log_metadata("XGBoost Version", xgboost.__version__)
+    self.run.log_model("XGBoost")
+    self.run.log_metadata("_IS_XGB", 'True')
+    self.run.log_metadata("Dataset columns", self.dtrain.num_col())
+    self.run.log_metadata("Dataset rows", self.dtrain.num_row())
+    self.run.log_metadata("Objective", self.params['objective'])
+    if self.validation_sets:
+      self.run.log_metadata("Number of Test Sets", len(self.validation_sets))
+
+  def log_params(self):
+    # set and log params, making sure to cross-reference XGB aliases
+    for key, value in self.params.items():
+      log_value = copy.deepcopy(value)  # overkill for most things but this value may be a list for whatever reason
+      if isinstance(log_value, (list, bool)):
+        log_value = str(log_value)
+      if key in XGB_ALIASES.keys():
+        setattr(self.run.params, XGB_ALIASES[key], log_value)
```

Comment:
Ok. I know we said that we want to log the non-alias param names, but I now think it's better to log the alias as is. We would just interpret the alias on the `sigopt-api` side when we are doing Experiments. So it's not jarring for the user to log `learning_rate` and see `eta` on the web.